muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/rest/ForumRESTService.java

```diff
@@ -517,7 +519,16 @@ public Response createThread(@PathParam ("AREAID") Long areaId, ForumThreadRESTM
     }
 
     if (sessionController.hasEnvironmentPermission(ForumResourcePermissionCollection.FORUM_WRITE_ENVIRONMENT_MESSAGES)) {
-      if (Boolean.TRUE.equals(newThread.getSticky()) || Boolean.TRUE.equals(newThread.getLocked())) {
+      LockForumThread lock = newThread.getLock() != null ? LockForumThread.valueOf(newThread.getLock()) : null;
+
+      if (Boolean.TRUE.equals(newThread.getSticky()) || lock != null) {
+        if (userEntityController.isStudent(sessionController.getLoggedUserEntity())) {
+          lock = null;
+        }
+        if (lock == LockForumThread.STUDENTS && userEntityController.isStudent(sessionController.getLoggedUserEntity())){
```

Comment:
Is this redundant check? Wouldn't `lock` be nulled in the previous if block? If so, I would remove the previous one and just give bad request / forbidden when the parameters are not allowed.